jre/java/java/util/stream/DoubleStream.java

```diff
@@ -147,19 +147,35 @@ public boolean tryAdvance(DoubleConsumer action) {
   }
 
   static DoubleStream iterate(double seed, DoubleUnaryOperator f) {
+    return iterate(seed, ignore -> true, f);
+  }
+
+  static DoubleStream iterate(double seed, DoublePredicate hasNext, DoubleUnaryOperator f) {
     Spliterator.OfDouble spliterator =
         new Spliterators.AbstractDoubleSpliterator(
             Long.MAX_VALUE, Spliterator.IMMUTABLE | Spliterator.ORDERED) {
+          private boolean first = true;
```

Comment:
isFirst and let's move before isTerminated

Reply:
done